Downloading fails if filename contains special characters

**Metadata**
Version: 5.0.0
OS: Arch Linux
Shell: bash
Anime: Any containing special characters such as `:`

**Describe the bug**
When downloading any file, it seems that the filename to write is not sanitized properly, which can lead to it being rejected by ffmpeg.

The example below shows it rejecting files with `:` but there may be more possible.

**Steps To Reproduce**

1. Run `ani-cli --download`
2. Search for show (this example shows gup)
3. Choose any of the options containing a `:` character.

**Expected behavior**
Should download properly.

**Actual behavior**
`ffmpeg` command fails to download because it seems to reject filenames with special characters.

```txt
$ ani-cli --download
Checking dependencies...
anidb.app links fetched
[out#0/mp4 @ 0x55c272c060c0] Error opening output ./Girls &amp; Panzer: Saishuushou Part 3 Episode 1.mp4: Invalid argument
Error opening output file ./Girls &amp; Panzer: Saishuushou Part 3 Episode 1.mp4.
Error opening output files: Invalid argument
```

Line 300 workaround i figured out:

#PLAYING

# shellcheck disable=SC2086
# $1 = video link, $2 = video name, $3 = extra flags

download() {
    command -v "yt-dlp" >/dev/null && yt-dlp "$1" --no-skip-unavailable-fragments --fragment-retries infinite -N 16 -o "$download_dir/$( echo "$2" | tr -d  ":!?").mp4" $3 && return 0
    ffmpeg -extension_picky 0 -loglevel error -stats -i "$1" -c copy "$download_dir/$( echo "$2" | tr -d  ":!?").mp4" $3
}

--- a/ani-cli
+++ b/ani-cli
@@ -351,6 +351,7 @@ play() {
     _end=$(printf "%s" "$ep_no" | grep -Eo '(-1|[0-9]+(\.[0-9]+)?)$')
     [ "$_start" = "-1" ] && ep_no=$(printf "%s" "$ep_list" | tail -n 1) && unset _start
     [ -z "$_end" ] || [ "$_end" = "$_start" ] && unset _start _end
+    [ "$_start" = "0" ] && _start=$(printf "%s" "$ep_list" | head -n 1)
     [ "$_end" = "-1" ] && _end=$(printf "%s" "$ep_list" | tail -n 1)
     _line_count=$(printf "%s\n" "$ep_no" | wc -l | tr -d "[:space:]")
     if [ "$_line_count" != 1 ] || [ -n "$_start" ]; then

--- a/ani-cli
+++ b/ani-cli
@@ -161,7 +161,7 @@ anidb_search() {
         [ "$curl_exe" = "curl" ] && die "Blocked by cloudflare. Try installing curl-impersonate"
         die "Blocked by cloudflare."
     fi
-    printf "%s" "$_page" | sed -nE 's|.*anime/([a-z0-9-]+-[0-9]+)".*alt="([^"]+)".*|\1\t\2|p' | sed -e "s|&#039;|'|g" -e 's|&quot;|"|g'
+    printf "%s" "$_page" | sed -nE 's|.*anime/([a-z0-9-]+-[0-9]+)".*alt="([^"]+)".*|\1\t\2|p' | sed -e "s|&#039;|'|g" -e 's|&quot;|"|g' -e "s|&amp;|\&|g"
 }
 
 # extract mal_id and seasons metadata

--- a/ani-cli
+++ b/ani-cli
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-version_number="5.0.2"
+version_number="5.0.3"
 
 # UI
 

--- a/ani-cli
+++ b/ani-cli
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-version_number="5.0.0"
+version_number="5.0.2"
 
 # UI
 

--- a/ani-cli.1
+++ b/ani-cli.1
@@ -17,7 +17,7 @@ This tool scrapes the site anidb.
 .SH OPTIONS
 .TP
 \fB\-e | --episode | -r | --range\fR \fI\,<episode>\/\fR
-Specify the episode numbers to watch. If range is specified it should be quoted or separated by a non-numeric character (eg. -).
+Specify the episode numbers to watch. If a range is specified, it should be quoted or separated by a non-numeric character (e.g. -). Use 0 for the first available episode and -1 for the last.
 .TP
 \fB\-c | --continue\fR
 Continue watching anime from history.

